### ISSUE TITLE
SVG image dimensions

### ISSUE THREAD

**Issue by seamusleahy** [Compass/compass#686]
The replace-text-with-dimensions() function doesn't work for SVGs because the image dimension function don't support SVG.

I was able to get it to work for my use case by adding a get_size_for_svg method to the ImageProperties class in Compass::SassExtensions::Functions::ImageSize.

```
def get_size_for_svg
  require 'rexml/document'
  svg = File.read(@file)
  doc = REXML::Document.new(svg)
  [doc.root.attributes['width'].to_i, doc.root.attributes['height'].to_i]     
end
```

I am reading the width and height attributes from the root <svg> element.

**Comment by chriseppstein** [comment]
Are pixels the only legal values of `width` and `height`? Pretty sure they're not. We would need to decide how to handle that case.

**Comment by seamusleahy** [comment]
You are correct in that it can be other values than pixel. Is it possible to return non pixel values?
- [svg width](http://www.w3.org/TR/SVG/struct.html#SVGElementWidthAttribute)
- [SVG length](http://www.w3.org/TR/SVG/types.html#DataTypeLength)
- [CSS units](http://www.w3.org/TR/2008/REC-CSS2-20080411/syndata.html#length-units)

**Comment by chriseppstein** [comment]
yes. it's possible to return any css unit with a little refactoring.

**Comment by waynedpj** [comment]
has there been any progress with this?  it seems that SVGs are still not supported by the image-width/image-height helpers as of Compass 0.12.2 (Alnilam).  i would be glad to help with testing/debugging.

thanks.

**Comment by akella** [comment]
Just wanted to bring this up. Very useful feature, not working for now. As we are replacing our png-sprites with new vector graphics...

**Comment by izelnakri** [comment]
Any plans to implement this feature?

**Comment by ffuubarbuzz** [comment]
One more, who hopes this will be implemented soon

**Comment by Rendez** [comment]
+1 :thumbsup:

**Comment by x-yuri** [comment]
Same here.

**Comment by fabio-coolshop** [comment]
+1 :thumbsup:

**Comment by andrewayala** [comment]
+1

**Comment by khoan** [comment]
two heads are better than one. There's a fix in https://github.com/sdsykes/fastimage/issues/49
